src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/NodeRunners.cs

```diff
@@ -210,15 +276,34 @@ public static IFullNodeBuilder MockIBD(this IFullNodeBuilder fullNodeBuilder)
                         ServiceDescriptor ibdService = services.FirstOrDefault(x => x.ServiceType == typeof(IInitialBlockDownloadState));
 
                         if (ibdService != null)
-                        {
                             services.Remove(ibdService);
-                            services.AddSingleton<IInitialBlockDownloadState, InitialBlockDownloadStateMock>();
-                        }
+
+                        services.AddSingleton<IInitialBlockDownloadState, InitialBlockDownloadStateMock>();
+                    });
+                }
+            });
+
+            return fullNodeBuilder;
+        }
+
+        public static IFullNodeBuilder MockRules(this IFullNodeBuilder fullNodeBuilder, IRuleRegistration ruleRegistration)
```

Comment:
Hu what? why do you need to replace rules?

Reply:
We cant mine x blocks with the ACTUAL miner as the block times are too short after each other, so consensus fails when we use  `GenerateStratisWithMiner`. This way we can inject the rules in cases where want to test mining 100 blocks for example but we dont want to wait 100 minutes.

Reply:
Hmm I think this is the wrong approach, in POW there is a flag to ignore difficulty, probably this is the same thing we need to do for POS how ever how exactly I am not sure.
We can add a flag to the code that checks the next target and reduce the target spacing requirements to be much lower, so effectively not have to wait so long between blocks.

@Aprogiena WDYT how can we achieve fast POS blocks in stratisregtest

Reply:
@dangershony pls ask me tomorrow and don't forget about it :)

Reply:
yeah good point, @fassadlr tomorrow I met apro we will discuss how we can make the POS mining faster while we still maintain the rules rules.

Reply:
@dangershony if we can get around this by changing the network's block time (or completely zero-ing) it in StratisRegTest, then this will work really great.

Reply:
So the solution is as following:
POS blocks expect a block every 64 seconds, so we can use a new DateTimeProvider and override the time to be 1 day ago, then every time a POS block is found the datetime will increment the current time bu 64 seconds and that will allow creating many POS blocks very fast.

So we should not remove any rules at all for this to work.

Reply:
> 1 day ago

Or more, it can even be one week or one month

Reply:
@dangershony OK, this is a good solution! I'll implement a custom DTP for this (the) tests. I will do this on this PR so that we keep record of the discussion.